include/awkward/array/RecordArray.h

```diff
@@ -441,7 +447,7 @@ namespace awkward {
     /// @brief See #recordlookup.
     const util::RecordLookupPtr recordlookup_;
     /// @brief See #length.
-    int64_t length_;
+    int64_t length_ = 0;
```

Comment:
Same as above: `length_` needs to always be set by a constructor, never assumed to be `0` if that somehow gets missed.